src/components/IconButton/IconButton.st.css

```diff
@@ -0,0 +1,71 @@
+:import {
+    -st-from: "wix-ui-core/index.st.css";
+    -st-named: ButtonNext;
+}
+
+:import {
+    -st-from: "../../common/formatters.st";
+    -st-named: color, fallback, font, applyOpacity;
+}
+
+:vars {
+    /*Defaults*/
+    DefaultColor: color-5;
+    DisabledColor: color-3;
+}
+
+:vars {
+    /*Overrides*/
+    IconColor: --overridable;
+}
+
+.root {
+    -st-extends: ButtonNext;
+    -st-states: disabled, skin(enum(line, full));
+    /*css goes here*/
+    background-color: transparent;
+    text-decoration: none;
+    border: none;
+    display: inline-block;
+    padding: 0;
+    margin: 0;
+    line-height: 0;
+}
+
+.root:hover {
+    opacity: 0.6;
+}
+
+.root:disabled {
+    cursor: default;
+}
+
+.root:disabled:hover {
+    opacity: 1;
+}
+
+.root:disabled .icon path {
+    stroke: color(value(DisabledColor));
```

Comment:
This works for the `'full` disabled, but not for `line` for some reason.
Maybe the icons aren't good.
I think it's because you have the `.root:skin(line) .icon path` rule below.
Maybe if you put it before it will work?

Reply:
or add a rule for `.root:skin(line):disabled .icon path`